keyboards/dm9records/plaid/keymaps/stephen-huan/keymap.c

```diff
@@ -0,0 +1,356 @@
+
+          double p = (readPin(led)) ? x : y;
+          if (rand() < p*RAND_MAX) {
+            togglePin(led);
+          }
+
+          /* following code is restriced to x + y <= 1
+           * if(rand() % 2 == 1) { */
+          /*   if(rand() % 2 == 0) { */
+          /*     writePinLow(led); */
+          /*   } */
+          /*   else { */
+          /*     writePinHigh(led); */
+          /*   } */
+          /* } */
+        }
+        break;
+      case LEDMODE_KEY:
+        if (record->event.pressed) {
+          writePinHigh(led);
+          return;
+        }
+        else {
+          writePinLow(led);
+          return;
+        }
+        break;
+      case LEDMODE_ENTER:
+        if (keycode==KC_ENT) {
+          writePinHigh(led);
+        }
+        else {
+          writePinLow(led);
+        }
+        break;
+
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  /* If the either led mode is keypressed based, call the led updater
+     then let it fall through the keypress handlers. Just to keep 
+     the logic out of this procedure */
+  if (led_config.red_mode >= LEDMODE_MODS && led_config.red_mode <= LEDMODE_ENTER) {
+      led_keypress_update(LED_RED, led_config.red_mode, keycode, record);
+  }
+  if (led_config.green_mode >= LEDMODE_MODS && led_config.green_mode <= LEDMODE_ENTER) {
+      led_keypress_update(LED_GREEN, led_config.green_mode, keycode, record);
+  }
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case LED_1:
+      if (record->event.pressed) {
+        if (led_config.red_mode==LEDMODE_ON) {
+            led_config.red_mode=LEDMODE_OFF;
+            writePinLow(LED_RED);
+        }
+        else {
+            led_config.red_mode=LEDMODE_ON;
+            writePinHigh(LED_RED);
+        }
+      }
+      break;
+    case LED_2:
+      if (record->event.pressed) {
+        if (led_config.green_mode==LEDMODE_ON) {
+            led_config.green_mode=LEDMODE_OFF;
+            writePinLow(LED_GREEN);
+        }
+        else {
+            led_config.green_mode=LEDMODE_ON;
+            writePinHigh(LED_GREEN);
+        }
+      }
+      break;
+    case LED_3:
+      led_config.red_mode=LEDMODE_MODS;
+      break;
+    case LED_4:
+      led_config.green_mode=LEDMODE_MODS;
+      break;
+    case LED_5:
+      led_config.red_mode=LEDMODE_BLINKIN;
+      break;
+    case LED_6:
+      led_config.green_mode=LEDMODE_BLINKIN;
+      break;
+    case LED_7:
+      led_config.red_mode=LEDMODE_KEY;
+      break;
+    case LED_8:
+      led_config.green_mode=LEDMODE_KEY;
+      break;
+    case LED_9:
+      led_config.red_mode=LEDMODE_ENTER;
+      break;
+    case LED_0:
+      led_config.green_mode=LEDMODE_ENTER;
+      break;
+  }
+
+  if (keycode >= LED_0 && keycode <= LED_9) {
```

Comment:
Sorry, my fault.  Looking at the enum, this should be: 
```suggestion
  if (keycode >= LED_1 && keycode <= LED_0) {
```

As 1 is the first in the enum, and 0 is the last.

Reply:
No problem, it's equally my fault that I didn't catch that. Committed the changes.